qlkube/src/decorateBaseSchema.js

```diff
@@ -0,0 +1,65 @@
+const { gql } = require('apollo-server-core');
+const { extendSchema } = require('graphql/utilities');
+const { addResolversToSchema } = require('@graphql-tools/schema');
+
+/**
+ *
+ * @param {*} targetQuery : Query wrapped int the extended object Type
+ * @param {*} extendedType : Object type that should be extended
+ * @param {*} argsNeeded : Optional arguments extracted from the father and passed from the wrapper
+ * @param {*} baseSchema : GraphQL Schema where the type and query are written
+ * @param {*} nameWrapper : Optional custom name for the wrapper
+ * @returns
+ */
+
+function capitalizeType(name) {
+  return name[0].toUpperCase() + name.slice(1);
+}
+
+module.exports = {
+  decorateBaseSchema: function (
+    targetQuery,
+    extendedType,
+    argsNeeded,
+    nameWrapper,
+    baseSchema
+  ) {
+    if (!targetQuery) return baseSchema;
+    if (!extendedType) return baseSchema;
+    const targetType = baseSchema.getQueryType().getFields()[targetQuery];
+    if (!targetType) return baseSchema;
+
+    nameWrapper = nameWrapper ? nameWrapper : 'fieldWrapper';
+    let typeWrapper = capitalizeType(nameWrapper);
+    let typeTargetQuery = capitalizeType(targetQuery);
+    const extension = gql`
+            extend type ${extendedType} {
+                ${nameWrapper}: ${typeWrapper}
+            }
+            type ${typeWrapper} {
+                ${targetQuery}: ${typeTargetQuery}
+            }
+        `;
+    const resolvers = {
+      [extendedType]: {
+        [nameWrapper]: (parent, args, context, info) => {
+          let newParent = {};
+          for (e of argsNeeded) {
+            newParent[e] = parent[e];
+          }
+          return newParent !== {} ? newParent : parent; // gestire errori
```

Comment:
The `!== {}` check will always result false, if you need this check you should use a loop with Obejct.keys. But I believe that a better approach would be to throw an error if the needed prop doesn't exist on the parent.

(also please remove the `// gestire errori` part)

Reply:
Being that argsNeeded is optional also newParent could be empty, isn't so?
May be `Object.keys(newParent).length !==0 ? newParent : parent` a good alternative? 
I could add also a default value for the optional agrs.